File: src/MixNetFx/FacFunctions/Dialogs/GenericFormSettings.vb
Imports Activedev
Imports System.Windows.Forms

Public Enum InfoItemFormEditMode
    View
    Edit
    AddNew
End Enum

Public Class InfoItemMaintenanceDialogResult

    Private myDialogResult As DialogResult
    Private myInfoItem As IInfoItem

    Sub New(ByVal InfoItem As IInfoItem, ByVal DialogResult As DialogResult)
        myDialogResult = DialogResult
        myInfoItem = InfoItem
    End Sub

    Public ReadOnly Property InfoItem() As IInfoItem
        Get
            Return myInfoItem
        End Get
    End Property

    Public ReadOnly Property DialogResult() As DialogResult
        Get
            Return myDialogResult
        End Get
    End Property

End Class

File: src/NetFxCSharp/Facesso.Functions/Dialogs/GenericFormSettings.cs
using ActiveDev;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Xml.Linq;

namespace Facesso.Functions
{
    public enum InfoItemFormEditMode
    {
        View,
        Edit,
        AddNew,
    }

    public class InfoItemMaintenanceDialogResult
    {
        private DialogResult myDialogResult;
        private IInfoItem myInfoItem;
        public InfoItemMaintenanceDialogResult(IInfoItem InfoItem, DialogResult DialogResult)
        {
            myDialogResult = DialogResult;
            myInfoItem = InfoItem;
        }

        public IInfoItem InfoItem
        {
            get
            {
                return myInfoItem;
            }
        }

        public DialogResult DialogResult
        {
            get
            {
                return myDialogResult;
            }
        }
    }
}